src/fluidsynth.c

```diff
@@ -896,14 +896,25 @@ int main(int argc, char **argv)
     /* load the soundfonts (check that all non options are SoundFont or MIDI files) */
     for(i = arg1; i < argc; i++)
     {
-        if(fluid_is_midifile(argv[i]))
+#if defined(WIN32)
+        int u16_count = MultiByteToWideChar(CP_ACP, 0, argv[i], -1, NULL, 0);
+        char *u16_buf = malloc((size_t)u16_count * 2);
```

Comment:
Pls. use something like
```suggestion
        LPWSTR u16_buf = FLUID_ARRAY(*LPWSTR, u16_count);
```

Reply:
And pls. check for NULL. Same below.